src/openproduct/producttypen/serializers/validators.py

```diff
@@ -81,3 +82,19 @@ def __call__(self, value, serializer):
             validate_publicatie_dates(pub_start_datum, pub_eind_datum)
         except ValidationError as e:
             raise serializers.ValidationError(e.message)
+
+
+class DoelgroepUplValidator:
+    requires_context = True
+
+    def __call__(self, value, serializer):
+        upl = get_from_serializer_data_or_instance(
+            "uniforme_product_naam", value, serializer
+        )
+
+        doelgroep = get_from_serializer_data_or_instance("doelgroep", value, serializer)
+
+        try:
+            validate_uniforme_product_naam_constraint(upl, doelgroep)
+        except ValidationError as e:
+            raise serializers.ValidationError(e.message)
```

Comment:
In `PublicatieDateValidator` and `DoelgroepUplValidator`, it would be better to specify the field
```suggestion
            raise serializers.ValidationError({"doelgroep": e.message})
```